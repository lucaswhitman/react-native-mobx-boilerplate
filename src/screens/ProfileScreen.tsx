import { inject, observer } from 'mobx-react/native';
import React, { Component } from 'react';
import {
    Button,
    Image,
    ScrollView,
    Text
} from 'react-native';
import { translate } from '../i18n';
import styles from '../styles';

export const PROFILE_SCREEN_VIEW_MAIN = 'profileScreenViewMain';
export const PROFILE_SCREEN_TEXT_FIRSTNAME = 'profileScreenTextFirstname';

export interface IProfileScreenProps {
  SessionStore?: any;
  UserStore?: any;
}

@inject('SessionStore', 'UserStore')
@observer
export default class ProfileScreen extends Component<IProfileScreenProps> {

  public render(): JSX.Element {
    let firstName = '';
    let city = '';
    let state = '';
    try {
      firstName = this.props.UserStore.currentUser['first_name'];
      city = this.props.UserStore.currentUser['city'];
      state = this.props.UserStore.currentUser['state'];
    } catch (e) {
      console.log(e);
    }
    return (
      <ScrollView testID={PROFILE_SCREEN_VIEW_MAIN} contentContainerStyle={styles.container}>
        {this.renderImage()}
        <Text testID={PROFILE_SCREEN_TEXT_FIRSTNAME} style={[styles.heading, styles.headingOne]}>
          {firstName}
        </Text>
        <Text style={[styles.pText]}>
          {city}, {state}
        </Text>
        <Button
          onPress={() => this.props.SessionStore.logout()}
          title={translate('common.logOut')}
        />
      </ScrollView>
    );
  }

  private renderImage(): JSX.Element|undefined {
    let url = undefined;
    try {
      url = this.props.UserStore.currentUser['photos'][0]['urls']['full'];
    } catch (e) {
      console.log(e);
    }
    if(url !== undefined) {
      return (
        <Image
          style={styles.userThumb}
          source={{uri: url}}
        />
      );
    }
  }
}
