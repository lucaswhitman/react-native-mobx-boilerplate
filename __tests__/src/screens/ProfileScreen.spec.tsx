
import { observable } from 'mobx';
import React from 'react';
import { render, shallow } from 'react-native-testing-library';
import ProfileScreen, {
  PROFILE_SCREEN_TEXT_FIRSTNAME,
  PROFILE_SCREEN_VIEW_MAIN
} from '../../../src/screens/ProfileScreen';
// import { UserStoreType } from '../../../src/stores/UserStore';

const currentUser = {
  first_name: 'Rahim',
  photos: [{
    urls: {
      full: 'http://localhost/ok.png'
    }
  }]
};

describe('ProfileScreen', () => {
  let SessionStore: any;
  let UserStore: any;

  beforeEach(() => {
    SessionStore = observable({
      logout: jasmine.createSpy()
    });
    UserStore = observable({
      currentUser: {}
    });
  });

  it('matches snapshot of shallow rendering', () => {
    const { output } = shallow(<ProfileScreen SessionStore={SessionStore} UserStore={UserStore} />);

    expect(output).toMatchSnapshot();
  });

  it('renders the profile info correctly', () => {
    UserStore.currentUser = currentUser;

    const { getByTestId } = render(<ProfileScreen SessionStore={SessionStore} UserStore={UserStore} />);

    // the scrollview is rendered
    const profileScreenViewMain = getByTestId(PROFILE_SCREEN_VIEW_MAIN);
    expect(profileScreenViewMain).not.toBeUndefined();

    // the profile view contains the proper first name
    const profileScreenTextFirstName = getByTestId(PROFILE_SCREEN_TEXT_FIRSTNAME);
    expect(profileScreenTextFirstName.props.children).toEqual(currentUser.first_name);
  });
});

