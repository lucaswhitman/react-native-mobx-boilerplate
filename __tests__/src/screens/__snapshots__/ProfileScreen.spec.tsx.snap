// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileScreen matches snapshot of shallow rendering 1`] = `
<ProfileScreen
  SessionStore={
    Object {
      "logout": [Function],
    }
  }
  UserStore={
    Object {
      "currentUser": Object {},
    }
  }
/>
`;
